### PR TITLE
Repo Card Updated,

### PR DESCRIPTION
**Pull request by SarveshChavan**
### Changes made
- Search button UI enhanced
- Alignment of text widgets in Repo Card
- Customizable Button Widget Added 
- This PR will remove the unnecessary usage of elevated buttons everywhere when required as it contains a custom button widget that we can customize according to need and it will remove the redundant codes.

### CODE REVIEW

**Comment by dk-a-dev** [comment]
@SarveshChavan i am looking into ur pr

**Comment by dk-a-dev** [comment]
@SarveshChavan there is an existing issue which connects contributors page to search page, another person is already working on that one if he does not fix these cards then only ur pr will be considered, or it will de dissolved!! No need to close it now I will do it later if I feel the issue was resolved with another big issue

**Comment by dk-a-dev** [comment]
This PR is on hold, I have reviewed it but I am waiting for contri page integration with search page- pr then will review again, just want to avoid clashes and conflicts

**Review by dk-a-dev — COMMENTED** [review]
PR on hold for now

**Comment by dk-a-dev** [comment]
@SarveshChavan what is is this custom button and where u planning to use it

**Comment by SarveshChavan** [comment]
The custom button will act as a widget. Rather than writing the sne elevated button code again and again (search button, view profile button, visit button, contributors button) we can call the custom button widget so that it will remove code redundancy. And also I have made it adaptive such that if icon needed in button then only pass it and also one can specify the height and width of the button also. 
Give me time till 11 pm today I'll take pull of currently updated main branch and I'll try to provide the use of custom button in my next commit in this PR

**Comment by dk-a-dev** [comment]
> The custom button will act as a widget. Rather than writing the sne elevated button code again and again (search button, view profile button, visit button, contributors button) we can call the custom button widget so that it will remove code redundancy. And also I have made it adaptive such that if icon needed in button then only pass it and also one can specify the height and width of the button also. Give me time till 11 pm today I'll take pull of currently updated main branch and I'll try to provide the use of custom button in my next commit in this PR

K got it current changes are reviewed, u can work further, I am thinking to create entirely a new issue for ur work
@SarveshChavan

**Comment by SarveshChavan** [comment]
I guess extending the scope of this issue is fine, and I'll file a new commit asap so once its done review it and let me know if anything needed

**Comment by SarveshChavan** [comment]
check the latest commit I have done it has no conflicts and also it reduces code redundancy

File: lib/screens/search_repos.dart [hunk 1]
```diff
@@ -7,6 +7,7 @@ import 'package:hacktoberfest_flutter/screens/settings.dart';
 import 'package:hacktoberfest_flutter/shared/colors.dart';
 import 'package:hacktoberfest_flutter/widgets/new_repo_card.dart';
 import 'package:http/http.dart' as http;
+import '../widgets/custom_button.dart';
```
Comment (dk-a-dev) [inline lib/screens/search_repos.dart:10]: Line 9-10 
> 9) import 'package:hacktoberfest_flutter/widgets/new_repo_card.dart';
> 10) import 'package:http/http.dart' as http;

File: lib/widgets/custom_button.dart [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                width: 10,
+              ),
+            Text(
+              buttonText,
+              textAlign: TextAlign.center,
+              style: const TextStyle(
+                  color: Colors.white,
+                  fontSize: 16,
+                  fontWeight: FontWeight.w500,
+                  overflow: TextOverflow.fade),
```
Comment (dk-a-dev) [inline lib/widgets/custom_button.dart:67]: add a comma "trailing comma after fade"
quickly commit

**Review by dk-a-dev — APPROVED** [review]
Good Job buddy it's almost ready just minor fixes i added

**Comment by dk-a-dev** [comment]
@SarveshChavan quick fixes remaining!! READY TO MERGE

**Comment by SarveshChavan** [comment]
> @SarveshChavan quick fixes remaining!! READY TO MERGE

all changes done

**Pull request merged**